File: app/src/test/java/com/aticosoft/appointments/mobile/business/test/common/infrastructure/persistence/TestPersistenceConfigurer.java
package com.aticosoft.appointments.mobile.business.test.common.infrastructure.persistence;

import com.aticosoft.appointments.mobile.business.infrastructure.persistence.PersistenceConfigurer;

import java.util.Map;

/**
 * Created by rodrigo on 22/08/15.
 */
public class TestPersistenceConfigurer extends PersistenceConfigurer {

    public static final String IN_MEMORY_LOCATION = "mem:db1";

    protected TestPersistenceConfigurer(Services services) {
        super(services);
    }

    @Override protected String location() {
        return IN_MEMORY_LOCATION;
    }

    @Override protected Map overrideConnectionUrlParameters(Map parameters) {
        parameters.remove(ConnectionUrlParameters.FILE_LOCK);
        parameters.remove(ConnectionUrlParameters.PAGE_SIZE);
        parameters.remove(ConnectionUrlParameters.CACHE_SIZE);
        return parameters;
    }
}

File: app/src/test/java/com/aticosoft/appointments/mobile/business/test/common/infrastructure/persistence/TestPersistenceConfigurer.kt
package com.aticosoft.appointments.mobile.business.test.common.infrastructure.persistence

import com.aticosoft.appointments.mobile.business.infrastructure.persistence.PersistenceConfigurer
import com.aticosoft.appointments.mobile.business.infrastructure.persistence.PersistenceConfigurer.ConnectionUrlParameters

/**
 * Created by rodrigo on 10/09/15.
 */
class TestPersistenceConfigurer(services: PersistenceConfigurer.Services) : PersistenceConfigurer(services) {

    private companion object {
        val IN_MEMORY_LOCATION: String = "mem:db1"
    }

    override protected fun location() = IN_MEMORY_LOCATION

    override protected fun overrideConnectionUrlParameters(parameters: MutableMap<String, Any>): Map<String, Any> {
        with(parameters) {
            remove(ConnectionUrlParameters.FILE_LOCK)
            remove(ConnectionUrlParameters.PAGE_SIZE)
            remove(ConnectionUrlParameters.CACHE_SIZE)
        }
        return parameters
    }
}
